cypress/full/modals_page_spec.ts

```diff
@@ -0,0 +1,629 @@
+      const btnChangeWidth = 'Change width';
+      const width300px = '300px';
+      const width800px = '800px';
+      const modalClassSM = '.modal-dialog.modal-sm';
+      const modalClassLG = '.modal-dialog.modal-lg';
+
+      it('example contains the button "Create template modal"', () => {
+        modals.isButtonExist(classChangeDemo, btnText);
+      });
+
+      it(`when user clicks on "Create template modal" button then modal is opened, "Change width" button
+       is present. the modal has width 300px and class "modal-dialog modal-sm"`,
+        () => {
+          modals.clickByText(classChangeDemo, btnText);
+          modals.isModalVisible(modals.modalContainer, true);
+          modals.isModalBtnExist(btnChangeWidth, 1);
+          modals.isModalWindowWidth(modals.modalContainer, width300px);
+          modals.isModalHasClass(modalClassSM);
+        });
+
+      it(`when user click on the button "Change width" then width of the modal is changed to 800px and
+      the modal class changed to "modal-dialog modal-lg"`, () => {
+        modals.clickByText(classChangeDemo, btnText);
+        modals.isModalVisible(modals.modalContainer, true);
+        modals.clickOnModalBtn(btnChangeWidth);
+        modals.isModalWindowWidth(modals.modalContainer, width800px);
+        modals.isModalHasClass(modalClassLG);
+      });
+    });
+  });
+
+  describe('Modals demo page test suite', () => {
+    beforeEach(() => modals.navigateTo());
+
+    describe('Directive examples', () => {
+      describe('Static modal', () => {
+        beforeEach(() => modals.scrollToMenu('Static modal'));
+
+        const staticModalDemo = modals.exampleDemosArr.directiveStatic;
+        const btnText = 'Static modal';
+        const btnX = '×';
+
+        it('example contains the button "Static modal"', () => {
+          modals.isButtonExist(staticModalDemo, btnText);
+        });
+
+        it(`when user clicks on "Static modal" button then modal is opened. after click on backdrop the modal
+       stays opened`, () => {
+          modals.clickByText(staticModalDemo, btnText);
+          modals.isModalVisible(modals.modalDialog, true);
+          modals.clickOnBackdrop();
+          modals.isModalVisible(modals.modalDialog, true);
+        });
+
+        it('when user clicks on the cross button then the modal is closed', () => {
+          modals.clickByText(staticModalDemo, btnText);
+          modals.isModalVisible(modals.modalDialog, true);
+          modals.clickOnDirectModalBtn(staticModalDemo, modals.modalHeader, btnX);
+          modals.isModalNotEnabled(modals.modalDialog, true);
+        });
+      });
+
+      describe('Optional sizes', () => {
+        beforeEach(() => modals.scrollToMenu('Optional sizes'));
+
+        const sizeDemo = modals.exampleDemosArr.directiveSizes;
+        const btnLargeModal = 'Large modal';
+        const btnSmallModal = 'Small modal';
+        const width300px = '300px';
+        const width800px = '800px';
```

Comment:
in this describe better to set cy.viewport with appropriate width and heigth

Reply:
added to the `beforeEach()`